bin/ruby-build

```diff
@@ -229,6 +229,23 @@ compute_sha2() {
   fi
 }
 
+compute_sha2_512() {
+  local output
+  if type shasum &>/dev/null; then
+    output="$(shasum -a 512 -b)" || return 1
+    echo "${output% *}"
+  elif type openssl &>/dev/null; then
+    local openssl="$(command -v "$(brew --prefix openssl 2>/dev/null || true)"/bin/openssl openssl | head -1)"
+    output="$("$openssl" dgst -sha512 2>/dev/null)" || return 1
+    echo "${output##* }"
+  elif type sha512sum &>/dev/null; then
+    output="$(sha512sum --quiet)" || return 1
```

Comment:
It's included in GNU coreutils: http://www.gnu.org/software/coreutils/manual/html_node/sha2-utilities.html#sha2-utilities